src/toil/cwl/cwltoil.py

```diff
@@ -2353,37 +3053,42 @@ def set_secondary(fileobj):
                 discover_secondaryFiles=True,
             )
 
-            def path_to_loc(obj):
-                if "location" not in obj and "path" in obj:
-                    obj["location"] = obj["path"]
-                    del obj["path"]
+            # Define something we can call to import a file and get its file
+            # ID.
+            file_import_function = functools.partial(
+                toil.importFile,
+                symlink=True
+            )
 
-            def import_files(inner_tool):
-                visit_class(inner_tool, ("File", "Directory"), path_to_loc)
-                visit_class(
-                    inner_tool, ("File",), functools.partial(add_sizes, fs_access)
-                )
-                normalizeFilesDirs(inner_tool)
-
-                adjustFileObjs(
-                    inner_tool,
-                    functools.partial(
-                        uploadFile,
-                        functools.partial(toil.importFile, symlink=True),
-                        fileindex,
-                        existing,
-                        skip_broken=True,
-                    ),
+            # Import all the input files, some of which may be missing optional
+            # files.
+            import_files(
+                file_import_function,
+                fs_access,
+                fileindex,
+                existing,
+                initialized_job_order,
+                skip_broken=True
+            )
+            # Import all the files associated with tools (binaries, etc.).
+            # Not sure why you would have an optional secondary file here, but
+            # the spec probably needs us to support them.
+            visitSteps(
+                tool, functools.partial(
+                    import_files,
+                    file_import_function,
+                    fs_access,
+                    fileindex,
+                    existing,
+                    skip_broken=True
                 )
+            )
 
-            # files with the 'file://' uri are imported into the jobstore and
-            # changed to 'toilfs:'
-            import_files(initialized_job_order)
-
-            visitSteps(tool, import_files)
-
-            for job_name, job_params in initialized_job_order.items():
-                rm_unprocessed_secondary_files(job_params)
+            for param_name, param_value in initialized_job_order.items():
```

Comment:
Hmmm... why was this changed from `job_name, job_params` to something more ambiguous?  Are the `initialized_job_order.items()` not all jobs?

Reply:
Nope! This isn't an "order" as in a sequence of jobs; it's an "order" as in instructions to run the jobs with certain arguments. I changed it to stop being misleading.